addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-mount-service.js

```diff
@@ -783,16 +789,22 @@ class EnvironmentMountService extends Service {
     return permissions;
   }
 
-  _prepareS3Mounts(studyInfo) {
+  async _prepareS3Mounts(studyInfo) {
     let mounts = [];
     if (studyInfo.length) {
+      const studyDataKmsKeyArn = this.settings.get(settingKeys.studyDataKmsKeyArn);
+
       // There might be multiple resources. In the future we may flatMap, for now...
       mounts = studyInfo.reduce(
         (result, { id, resources, category, writeable }) =>
           result.concat(
             resources.map(resource => {
               const { bucket, prefix } = parseS3Arn(resource.arn);
-              return { id, bucket, prefix, category, writeable };
+              const mount = { id, bucket, prefix, category, writeable };
+              if (category !== 'Open Data') {
+                mount.kmsKeyId = studyDataKmsKeyArn;
+              }
```

Comment:
The `getStudyAccessInfo` was not adding the KMS key arn to the mounts. This is not a bug but the kms key arn will be required when we work on BYOB. Since I was in this part of the code I quickly added this code and tested the generated mounts manually. There are no existing tests for that method. Adding test for this case would first require adding tests for that method in general.